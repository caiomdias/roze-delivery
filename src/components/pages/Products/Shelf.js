import React, { useEffect, useState } from 'react'
import ProductCard from '../../common/productCard/ProductCard'
import { getProducts } from '../../../services/ZeApi'
import productNotFound from '../../../assets/images/png/product-not-found.png';


const Shelf = (props) => {
	const { categoryId, distributorId } = props
	const [ categoryIdState, setCategoryIdState ] = useState(categoryId || '') 
	const [ distributorIdState, setDistributorId ] = useState(distributorId || '')
	const [ products, setProducts ] = useState([])

	const variables = {
		id: distributorIdState,
		search: '',
		categoryId: categoryIdState
  }
  
  useEffect(() => {
    if(distributorIdState && categoryIdState) {
      getProducts(variables)
        .then(({ data }) => {
          if(data.length != 0) {
            setProducts(data.poc.products)
          }
        })
    }
  }, [])

	return (
    <div className="shelf">
      {
        products.length !== 0
          ? products.map((item, index) => (
            <ProductCard key={index} product={item}/>
          ))
          : <div className="shelf__not-found">
              <img src={productNotFound}></img>
              <p>Putz, n√£o conseguimos encontrar nenhum produto dessa categoria</p>
            </div>
      }
    </div>
	)
};

Shelf.defaultProps = {
	categoryId: '',
	distributorId: '',
}

export default Shelf;
