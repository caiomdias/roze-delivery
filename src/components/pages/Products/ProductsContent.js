import React, { useEffect, useState } from 'react';
import Shelf from './Shelf';
import { getCategories } from '../../../services/ZeApi'

const ProductsContent = ({ distributorId }) =>  {
  const [ allCategories, setAllCategories ] = useState([])
  const [ distributorIdState, setDistributorIdState ] = useState(distributorId || '')

  const variables = {
    id: distributorIdState,
    search: '',
    categoryId: null
  }

  useEffect(() => {
    getCategories(variables)
      .then(({ data }) => {
        if(data.allCategory.length != 0) {
          setAllCategories(data.allCategory)
        }
      })
    return;
  }, [])

  return (
    <div className="products-content">
      <div className="container">
        {
          allCategories
          ? allCategories.map((item, index) =>
            (
            <div key={index} className="item">
              <p>{item.title}</p>
              <Shelf categoryId={item.id} distributorId={distributorIdState}/>
            </div>
            ))
          : ''
        }
      </div>
    </div>
  );
  };

ProductsContent.defaultProps = {
  distributorId: ''
}

export default ProductsContent;
