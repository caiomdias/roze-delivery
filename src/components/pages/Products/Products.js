import React, { useEffect, useState } from 'react';
import Header from '../../common/header/Header';
import ProductsContent from './ProductsContent';
import shortLogo from '../../../assets/images/png/logo.png';
import { getDistributor } from '../../../services/ZeApi'

const headerConfig = {
  image: shortLogo,
  hasSearch: true,
  style: {
    width: '55px',
    cursor: 'pointer',
  }
};

const Products = (props) => {
  const { lat, long } = props.history.location.state
  const [distributorId, setDistributorId] = useState('532')
  const [latState, setLatState] = useState(lat || '')
  const [longState, setLongState] = useState(long || '')

  const variables = {
    algorithm: "NEAREST",
    lat: latState,
    long: longState,
    now: new Date()
  }

  useEffect(() => {
    if(latState, longState) {
      getDistributor(variables)
      .then(({ data }) => {
        if(data.pocSearch.length != 0) {
          setDistributorId(data.pocSearch[0].id)
        }
      })
    }
    return;
  }, [])

  return (
    <div>
      <Header config={headerConfig}/>
      <ProductsContent distributorId={distributorId}/>
    </div>
  )
};

Products.defaultProps = {
  lat: '',
  long: '',
}


export default Products;