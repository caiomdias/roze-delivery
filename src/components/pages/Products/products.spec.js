import React from 'react';
import renderer from 'react-test-renderer';
import Products from './Products';

jest.mock('react-router-dom', () => ({
    useLocation: jest.fn().mockReturnValue({
      pathname: '/',
      search: '',
      hash: '',
      state: null,
      key: '5nvxpbdafa',
	}),
	useHistory: jest.fn().mockReturnValue({
		push: jest.fn().mockReturnValue()
	}),
}));

const props = {
	history: {
		location: {
			state: {
				lat: '111111',
				long: '22222'
			}
		},
	}
}

describe("Products Component", () => {
	it('Should render correctly Products component', () => {  
		const productsComponent = renderer.create(<Products {...props}/>).toJSON();
		expect(productsComponent).toMatchSnapshot();
	});
});