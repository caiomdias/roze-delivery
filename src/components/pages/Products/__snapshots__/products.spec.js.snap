// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Component Should render correctly Products component 1`] = `
<div>
  <header
    className="header"
  >
    <div
      className="container content"
    >
      <div
        className="header__logo"
        onClick={[Function]}
      >
        <img
          alt="Jose Entregas"
          src="test-file-stub"
          style={
            Object {
              "cursor": "pointer",
              "width": "55px",
            }
          }
        />
      </div>
      <div
        className="header__search"
      >
        <div
          className="search"
        >
          <div
            className="search__input"
          >
            <div
              className="icon"
            />
            <input
              className="input"
              onChange={[Function]}
              placeholder="Pesquise sua bebida favorita"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    className="products-content"
  >
    <div
      className="container"
    />
  </div>
</div>
`;
